docs/source/platform/schema-validation.md

```diff
@@ -80,90 +123,40 @@ type Query {
 }
 ```
 
-GraphQL-aware client tooling, like GraphQL Playground and GraphiQL, use this information to assist developers in making the right choices.  These tools will:
+GraphQL-aware client tooling, like [Apollo VScode](./vscode.html), GraphQL Playground, and GraphiQL, use this information to help developers make the right choices. These tools will:
 
-* Provide developers with the helpful deprecation message referring them to the new name.
-* Avoid auto-completing the field.
+- Provide developers with the helpful deprecation message referring them to the new name.
+- Avoid auto-completing the field.
 
 Over time, usage will fall for the deprecated field and grow for the new field.
 
-> Using tools like [Apollo Engine](https://www.apollographql.com/platform), it’s possible to make educated decisions about when to retire a field based on actual usage data through schema analytics.
+> the Apollo Platform contains a [trace warehouse](./tracing.html) to enable educated decisions about when to retire a field based on usage data through schema analytics.
 
-<h3 id="non-breaking">Non-breaking changes</h3>
+<h3 id="versioning">What about versioning?</h3>
 
-Sometimes we want to keep a field, but change how clients use it by adjusting its variables. For example, if we had a `getUsers` query that we used to fetch user data based off of a list of user `ids`, but wanted to change the arguments to support a `groupId` to look up users of a group or filter the users requested by the `ids` argument to only return users in the group:
+Versioning is a technique to prevent necessary changes from becoming breaking changes. Developers who have worked with REST APIs past may have various patterns for versioning the API, commonly by using a different URI (e.g. `/api/v1`, `/api/v2`, etc.) or a query parameter (e.g. `?version=1`). With this technique, an application can easily end up with many different API endpoints over time, and the question of _when_ an API can be deprecated can become problematic. While version a GraphQL API the same way may be tempting, multiple graphql endpoints add exponential complexity to schema development and quickly becomes unmaintainable.
 
-```graphql
-type Query {
-  # what we have
-  getUsers(ids: [ID!]!): [User]!
+<h3 id="what-about-versioning">How about never make a breaking change?</h3>
 
-  # what we want to end up with
-  getUsers(ids: [ID!], groupId: ID!): [User]!
-}
-```
+To another extreme, teams could choose to avoid making any change that might break an operation, ignoring consumer usage. While a viable strategy for maintaining clients in the short term, this limits the flexibility of the schema. Checking changes against usage enables more improvements to API ergonomics, such as removing fields or default argument updates. Positive API experience leads to better developer experience and more robust client and server interaction.
```

Comment:
Can we make this sound a little bit less opinionated? I'm all for listing out the pros and cons, but this sentence is basically like "some people do this and they probably shouldn't" which would make me feel pretty bad if my company was doing this and had good reasons.